app/code/Magento/Ui/view/base/web/js/modal/modal-component.js

```diff
@@ -361,7 +361,44 @@ define([
 
             if (this.valid) {
                 this.closeModal();
+            } else {
+                this._focusInvalid(this.elems());
+            }
+        },
+
+        /**
+         * Focus to first invalid element
+         *
+         * @param {HTMLElement} elems
+         * @param {boolean} firstRun
+         * @returns {boolean}
+         */
+        _focusInvalid: function (elems, firstRun = true) {
+            var self = this;
```

Comment:
Please bind context to the function instead of use `self` variable. For that, you can use `bind` `call` or `apply` methods